src/ir/syntax.hpp

```
@@ -188,29 +188,50 @@ struct ArgPair {
+/// info)` produces the full ResolvedCall with name, support status, and
+/// contract. Sibling key-only call instructions are `Callx` (register-indirect)
+/// and `CallBtf` (pre-resolution kfunc).
 struct Call {
     int32_t func{};
     CallKind kind{CallKind::helper};
-    // Equality intentionally matches only functional identity (call source + id).
-    // Metadata such as is_supported/unsupported_reason is diagnostic-only.
-    constexpr bool operator==(const Call& other) const { return func == other.func && kind == other.kind; }
 
-    // TODO: move name and signature information somewhere else
+    constexpr bool operator==(const Call&) const = default;
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)